### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by olix0r**
- parallelize testing in circleci, cuts CI times in half
- use coveralls for coverage reporting: https://coveralls.io/github/BuoyantIO/linkerd
- test-config.sh touchups:
  - make POSIX compliant with /bin/sh (fixes failures in parallelized CI)
  - more robust waiting for linkerd startup, should reduce timeout-related failures.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -4,14 +4,14 @@ resolvers += "twitter-repo" at "https://maven.twttr.com"
 addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.6.0")
 
 // testing
-addSbtPlugin("org.scoverage"     % "sbt-scoverage" % "1.3.3")
+addSbtPlugin("org.scoverage"  % "sbt-scoverage" % "1.3.5")
+addSbtPlugin("org.scoverage"  % "sbt-coveralls" % "1.1.0")
```
Comment (adleong) [inline project/plugins.sbt]: need a blank line between lines because sbt

File: project/LinkerdBuild.scala [hunk 2]
```diff
@@ -184,6 +188,8 @@ object LinkerdBuild extends Base {
     val main = projectDir("namerd/main")
       .dependsOn(core, admin, configCore)
       .withBuildProperties()
+      .settings(coverageExcludedPackages := ".*")
+
```
Comment (adleong) [inline project/LinkerdBuild.scala]: extra blank line

**Comment by adleong** [comment]
:star: 👖 cool!

**Comment by esbie** [comment]
⭐  yay! ⭐

**Pull request merged**